Backend/services/auctionSystem.js

```diff
@@ -0,0 +1,222 @@
+const { startOfDay, endOfDay } = require("date-fns");
+
+class AuctionSystem {
+  constructor(prisma) {
+    this.prisma = prisma;
+    this.bids = [];
+    this.items = new Map();
+    this.assignments = new Map();
+  }
+
+  calculateBidScore(bid) {
+    const maxBid = Math.max(
+      ...this.bids.filter((b) => b.itemId === bid.itemId).map((b) => b.amount)
+    );
+    const normalizedBid = maxBid > 0 ? bid.amount / maxBid : 0;
+    const userBidsForTheDay = this.bids.filter(
+      (b) => b.bidder.id === bid.bidder.id
+    ).length;
+    const normalizedBidsPerDay = Math.min(userBidsForTheDay / 10, 1);
+    const normalizedTime = (Date.now() / 1000 - bid.timestamp) / 86400;
```

Comment:
could we add some comments as to how you chose 10 and 1000 - bid.timestamp and 86400?